packages/react-components/react-charts-preview/src/utilities/FocusableTooltipText.tsx

```diff
@@ -1,80 +1,66 @@
+  const tooltipChild = useRef<HTMLSpanElement>(null);
+  const async = useRef(new Async()).current;
+  const resizeObserver = useRef<ResizeObserver>();
 
-    this._async = new Async(this);
-  }
-
-  public render(): React.ReactNode {
-    const { className, content, accessibilityData } = this.props;
-
-    return (
-      <TooltipHost overflowMode={TooltipOverflowMode.Self} hostClassName={className} content={content}>
-        <span
-          {...getAccessibleDataObject(accessibilityData)}
-          ref={this._tooltipChild}
-          data-is-focusable={this.state.textOverflow}
-        >
-          {content}
-        </span>
-      </TooltipHost>
-    );
-  }
+  const getTargetElement = useCallback((): HTMLElement | undefined => {
+    if (!tooltipChild.current || !tooltipChild.current.parentElement) {
+      return undefined;
+    }
+    return tooltipChild.current.parentElement;
+  }, [tooltipChild]);
 
-  public componentDidMount(): void {
-    const overflowElement = this._getTargetElement();
-    if (window.ResizeObserver && overflowElement) {
-      this._resizeObserver = new ResizeObserver(this._async.debounce(this._checkTextOverflow, 500));
-      this._resizeObserver.observe(overflowElement);
+  const checkTextOverflow = useCallback(() => {
+    const overflowElement = getTargetElement();
+    const isTextOverflow = !!overflowElement && hasOverflow(overflowElement);
+    if (isTextOverflow !== textOverflow) {
+      setTextOverflow(isTextOverflow);
     }
-  }
+  }, [getTargetElement, textOverflow, setTextOverflow]);
 
-  public componentWillUnmount(): void {
-    this._resizeObserver?.disconnect();
-    this._async.dispose();
-  }
+  useEffect(() => {
+    checkTextOverflow();
+  }, [checkTextOverflow]);
 
-  private _checkTextOverflow = (): void => {
-    const overflowElement = this._getTargetElement();
-    const textOverflow = !!overflowElement && hasOverflow(overflowElement);
-    if (textOverflow !== this.state.textOverflow) {
-      this.setState({ textOverflow });
+  useEffect(() => {
+    const overflowElement = getTargetElement();
+    if (window.ResizeObserver && overflowElement) {
+      resizeObserver.current = new window.ResizeObserver(async.debounce(checkTextOverflow, 500));
+      resizeObserver.current.observe(overflowElement);
     }
-  };
 
-  private _getTargetElement = (): HTMLElement | undefined => {
-    if (!this._tooltipChild.current || !this._tooltipChild.current.parentElement) {
-      return undefined;
-    }
+    return () => {
+      resizeObserver.current?.disconnect();
+      async.dispose();
+    };
+  }, [async, checkTextOverflow, getTargetElement]);
 
-    return this._tooltipChild.current.parentElement;
-  };
-}
+  return (
+    <div className={props.className}>
+      <Tooltip content={props.content} relationship="label">
+        <span {...getAccessibleDataObject(props.accessibilityData)} ref={tooltipChild} data-is-focusable={textOverflow}>
```

Comment:
is this block accessible?

Reply:
the tooltip itself is not accessible but the text is accessible via keyboard.